In [1]:
# Importing the libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd
# Importing the dataset
dataset = pd.read_csv('Churn_Modelling.csv')
X = dataset.iloc[:, 3:13].values
y = dataset.iloc[:, 13].values
# Encoding categorical data
from sklearn.preprocessing import LabelEncoder, OneHotEncoder
from sklearn.compose import ColumnTransformer
# Contry
labelencoder_X_1 = LabelEncoder()
X[:, 1] = labelencoder_X_1.fit_transform(X[:, 1])
# Male/Female
labelencoder_X_2 = LabelEncoder()
X[:, 2] = labelencoder_X_2.fit_transform(X[:, 2])
onehotencoder = ColumnTransformer([("Geography", OneHotEncoder(), [1])], remainder =
'passthrough')
X = onehotencoder.fit_transform(X)
X = X[:, 1:]
# Splitting the dataset into the Training set and Test set
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.2,
random_state = 0)
# Feature Scaling
from sklearn.preprocessing import StandardScaler
sc = StandardScaler()
X_train = sc.fit_transform(X_train)
X_test = sc.transform(X_test)

In [2]:
# Part 2 - Now let's make the ANN!
# Importing the Keras libraries and packages

import keras
from keras.models import Sequential
from keras.layers import Dense
from keras.layers import Dropout
# Initialising the ANN
from scikeras.wrappers import KerasClassifier
from sklearn.model_selection import cross_val_score
from keras.models import Sequential
from keras.layers import Dense
# Building a function to construct an ANN
def build_classifier():
 classifier = Sequential()
 classifier.add(Dense(units = 6, kernel_initializer = 'uniform', activation =
'relu', input_dim = 11))
 classifier.add(Dense(units = 6, kernel_initializer = 'uniform', activation =
'relu'))
 classifier.add(Dense(units = 1, kernel_initializer = 'uniform', activation =
'sigmoid'))
 classifier.compile(optimizer = 'adam', loss = 'binary_crossentropy', metrics =
['accuracy'])
 return classifier
# calling the function
classifier = KerasClassifier(build_fn = build_classifier, batch_size = 5, epochs =
50)
classifier1 = KerasClassifier(build_fn = build_classifier, batch_size = 10, epochs =
100)
accuracies = cross_val_score(estimator = classifier, X = X_train, y = y_train, cv =
10, n_jobs = -1)
accuracies1 = cross_val_score(estimator = classifier1, X = X_train, y = y_train, cv =
10, n_jobs = -1)
# Evaluating, Improving and Tuning the ANN
# Evaluating the ANN
mean = accuracies.mean()
variance = accuracies.std()
mean1 = accuracies1.mean()
variance1 = accuracies1.std()
print(mean, variance)
print(mean1, variance1)

0.83925 0.017474982117301287
0.8455 0.021824584761227412


batch_size = 5, epochs = 50: mean 0.83925 / variance 0.017474982117301287

batch_size = 10, epochs = 100: mean 0.8455 / variance 0.021824584761227412